python/lsst/meas/astrom/pessimistic_pattern_matcher_b_3D.py

```diff
@@ -2,11 +2,49 @@
 from __future__ import division, print_function, absolute_import
 
 import numpy as np
+from scipy.optimize import leastsq
 from scipy.spatial import cKDTree
 
 import lsst.pipe.base as pipeBase
 
 
+def _rotation_matrix_chi_sq(flattened_rot_matrix, src_pattern, ref_pattern):
+    """Compute the squared differences for least squares fitting.
+
+    Given a flattened rotation matrix, a N point pattern from the source
+    catalog and the reference pattern the sources match to, compute return
+    the squared differences between the points in the two patterns after
+    rotation.
+
+    Parameters
+    ----------
+    flattened_rot_matrix : float array
+        A flattened array representing a 3x3 rotation matrix.
```

Comment:
I suggest you use a normal 2D array, instead. It makes the API easier to understand, and the matrix needs to be unflattened at some point -- might as well do it before calling this.

if you insist on the data being a 1-D array then you must explain the order in which the elements are stored.